pinot-server/src/main/java/org/apache/pinot/server/starter/helix/SegmentOnlineOfflineStateModelFactory.java

```diff
@@ -247,4 +250,12 @@ public void onBecomeDroppedFromError(Message message, NotificationContext contex
       }
     }
   }
+
+  @Override
+  public ExecutorService getExecutorService(String resourceName, String fromState, String toState) {
```

Comment:
The helix docs of StateModelFactory states "This method is called just once per transition type per resource. "

What is a transition type ? Will this be an issue for us ?